File: Betch-2-1/ClassWork/DataStructurs/Unit-1/Sorting/BubbleSort/BubbleSort.py
class BubbleSort:
    def __init__(self,a):
        self.a=a
        BubbleSort._Sort(self)
    @staticmethod  
    def _Sort(self):
        for i in range(len(a)):
            for j in range(len(a)-i-1):
                if a[j]>a[j+1]:
                    a[j],a[j+1]=a[j+1],a[j]
                    print(f"iteration:{i+1} res:{a}")
        print()
        print(f"final sorted result:{a}")
a=[20,3,17,19,25,35,9,42,16,27]
obj=BubbleSort(a)

File: Betch-2-1/ClassWork/DataStructurs/Unit-1/Sorting/BubbleSort/BubbleSort.java
class BubbleSort{
    private static int a[];
    BubbleSort(int[] a){
        this.a=a;
        BubbleSort.sort();
    }
    private static void sort(){
        for(int i=0;i<a.length;i++){
            for(int j=0;j<-i-1;j++){
                if(a[j]>a[j+1]){
                    a[j]+=a[j+1];
                    a[j+1]-=a[j];
                    a[j]+=a[j+1];
                    System.out.printf("iteration:%d res:%d",i,a);
                }
            }
       }
       //System.out.println(a);
    }
}

class Main{
    public static void main(String[] args) {
        int[] a={20,3,17,19,25,35,9,42,16,27};
        BubbleSort bs=new BubbleSort(a);
    }
}